src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -3,10 +3,22 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="Microsoft.Managed.Core.targets"/>
 
-  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
-                            ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1')">
-    <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
-    <LangVersion Condition="'$(LangVersion)' == ''">$(MaxSupportedLangVersion)</LangVersion>
+  <PropertyGroup>
+    <!-- .NETCoreApp < 3.0, .NETStandard < 2.1, or any other target framework -->
```

Comment:
> or any other target framework [](start = 47, length = 30)

I didn't understand this part ("any other target framework") and how it matches the logic below

Reply:
The logic is kinda weird to follow, but this will catch everything *except* .NETCoreApp >= 3 and .NETStandard >= 2.1 , so all other target frameworks get set here. The following checks only kick in when `_MaxSupportedLangVersion` isn't set so they won't cause it to change. I think we can probably simplify the logic below though. 

---
In reply to: [443010616](https://github.com/dotnet/roslyn/pull/44911#discussion_r443010616) [](ancestors = 443010616)

Reply:
Got it. Thanks
nit: consider inverting the condition with a top-level "not": `not (.NETCoreApp >= 3 || .NETStandard >= 2.1)`

---
In reply to: [445156354](https://github.com/dotnet/roslyn/pull/44911#discussion_r445156354) [](ancestors = 445156354,443010616)